lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -560,6 +569,10 @@ def _configure_network_attachment_definitions(self):
             self.kubernetes.create_network_attachment_definition(
                 network_attachment_definition=network_attachment_definition_to_create
             )
+        if nad_config_changed:
```

Comment:
I see that `nad_config_changed` is only set when we remove a `nad`, wouldn't this also be needed if we have a new `nad` to create? If we ever want to dynamically create `nads` that would be useful.

Reply:
According to observation, if a new NAD (which does not exist before) is created, it becomes active immediately. However, if we modify an existing NAD, the modification appears after pod restart. This is the reason of why we are restarting if there is NAD change.  The NAD change triggers both delete and create operations sequentially but delete operation is distinguishing feature for modifications. We restart the pod once because decreasing number of restart seems better (?) and enough. If we also restart pod for new nad creation, this causes one more pod restart  (which seems not necessary) just after we deploy app although it is using default MTU size.

Reply:
Ok, thanks for the clarification!